packages/java/endpoint/src/main/java/com/vaadin/hilla/route/RouteUnifyingIndexHtmlRequestListener.java

```diff
@@ -190,6 +169,38 @@ protected Map<String, AvailableViewInfo> collectServerViews() {
                         Function.identity()));
     }
 
+    private boolean hasMainMenu(DeploymentConfiguration deploymentConfiguration,
+            VaadinRequest request) {
+        Map<String, AvailableViewInfo> clientItems = MenuRegistry
+                .collectClientMenuItems(true, deploymentConfiguration, request);
```

Comment:
Probably the result of calling the `MenuRegistry.collectClientMenuItems` could be passed to this method to avoid a second call and loading/reading the json file again.

Reply:
done